features/step_definitions/download_endpoint_steps.rb

```diff
@@ -0,0 +1,28 @@
+require 'open-uri'
+require 'tempfile'
+
+Given(/^I try download "([^"]*)" from channel "([^"]*)"$/) do |rpm, channel|
+  url = "#{Capybara.app_host}/rhn/manager/download/#{channel}/getPackage/#{rpm}"
+  if @token
+    url = "#{url}?#{@token}"
+  end
+  puts url
+  Tempfile.open(rpm) do |tmpfile|
+    @download_path = tmpfile.path
+    begin
+      open(url) do |urlfile|
+        tmpfile.write(urlfile.read)
```

Comment:
>When a Tempfile object is garbage collected, or when the Ruby interpreter exits, its
>associated temporary file is automatically deleted. This means that’s it’s unnecessary to >explicitly delete a Tempfile after use, though it’s good practice to do so: not explicitly deleting
>unused Tempfiles
>can potentially leave behind large amounts of tempfiles on the filesystem until they’re
>garbage collected. The existence of these temp files can make it harder to determine a new
>Tempfile filename.

So, it is removed when garbage collected. It is a good practice to do it explicitly.

Finalizer is setup like this: https://github.com/ruby/ruby/blob/trunk/lib/tempfile.rb#L135